backend/sirius-components-diagrams/src/main/java/org/eclipse/sirius/components/diagrams/components/DiagramComponent.java

```diff
@@ -89,12 +89,19 @@ public Element render() {
         children.addAll(edges);
 
         // @formatter:off
+        Position position = optionalPreviousDiagram.map(Diagram::getPosition)
+                .orElse(Position.UNDEFINED);
+
+
+        Size size = optionalPreviousDiagram.map(Diagram::getSize)
+                .orElse(Size.UNDEFINED);
+
         DiagramElementProps diagramElementProps = DiagramElementProps.newDiagramElementProps(diagramId)
                 .targetObjectId(targetObjectId)
                 .descriptionId(diagramDescription.getId())
                 .label(label)
-                .position(Position.UNDEFINED)
-                .size(Size.UNDEFINED)
+                .position(position)
+                .size(size)
```

Comment:
This is what was causing the "1 pixel difference bug" since the position of the diagram was being rest to (-1, -1) every time the layout algorithm was applied.